guides/stable/3D_interactivity.md

```diff
@@ -0,0 +1,131 @@
+        If the click does not intersect the axis-aligned data bounding box,
+        None is returned.
+	"""
+```
+
+## Adding 3D interactivity via mouse events
+Custom 3D interactivity can be added via mouse callbacks. The `layer.get_ray_intersections()` function has been designed to work seamlessly with the napari mouse callback event. You can pass the mouse callback event properties to `layer.get_ray_intersections()` get the `start_point` and `end_point` of where the click line intersects the layer data. 
+
+```
+@layer.mouse_drag_callbacks.append
+def on_click(layer, event):
+	# get the points where the click intersects the 
+    start_point, end_point = layer.get_ray_intersections(
+        position=event.position,
+        view_direction=event.view_direction,
+        dims_displayed=event.dims_displayed,
+        world=True
+    )
+    if (start_point is not None) and (end_point is not None):
+    	# use start_point and end_point to interrogate layer data
+```
+
+For an example implementation, see the [`cursor_ray.py`](https://github.com/napari/napari/blob/master/examples/cursor_ray.py) example.
+
+## Getting the layer data under the cursor
+We are currently implementing convenience methods to the layers (`layer.get_value()`) to get the layer data value underneath the cursor that is "on top" (i.e., closest to `start_point`). Like `layer.get_ray_intersections()`, `layer.get_value()` takes the click position, view direction, dims_displayed in either world or layer coordinates (see `world` argument) as input. Thus, it can be easily integrated into a mouse event callback. Note that `layer.get_value()` returns `None` if the layer is not currently visible. See the docstring below for details.
+
+```
+    def get_value(
+        self,
+        position,
+        *,
+        view_direction: Optional[np.ndarray] = None,
+        dims_displayed: Optional[List[int]] = None,
+        world=False,
+    ):
+        """Value of the data at a position.
+
+        If the layer is not visible, return None.
+
+        Parameters
+        ----------
+        position : tuple
+            Position in either data or world coordinates.
+        view_direction : Optional[np.ndarray]
+            A unit vector giving the direction of the ray in nD world coordinates.
+            The default value is None.
+        dims_displayed : Optional[List[int]]
+            A list of the dimensions currently being displayed in the viewer.
+            The default value is None.
+        world : bool
+            If True the position is taken to be in world coordinates
+            and converted into data coordinates. False by default.
+
+        Returns
+        -------
+        value : tuple, None
+            Value of the data. If the layer is not visible return None.
+```
+
+We have implemented `layer.get_value()` for the layers below. All other layers with return `None` from `layer.get_value()` when called while the viewer is in a 3D rendering mode.
```

Comment:
```suggestion
We have implemented `layer.get_value()` for the layers below. All other layers with return `None` from `layer.get_value()` when called while the viewer is in a 3D rendering mode.
```
This sounds like not ideal. Maybe create an issue about what's happening here on development side at the moment and link the issue...

Reply:
issue added at napari/napari#3187

Reply:
```suggestion
Progress on the implementation of `layer.get_value()` can be found at [#3187](https://github.com/napari/napari/issues/3187). Layers for which this is not yet implemented will return `None` if `layer.get_value()` is called whilst the viewer is in a 3D rendering mode.
```